src/main/java/com/birdie/kafka/connect/smt/Outbox.java

```diff
@@ -0,0 +1,240 @@
+                            headers.add(k, v, Schema.STRING_SCHEMA);
+                        });
+                    } catch (JsonProcessingException e) {
+                        LOGGER.error("Could not decode headers.", e);
+                    }
+                    break;
+                default:
+                    LOGGER.error("Field 'headers' should be org.apache.kafka.connect.data.Schema.Type.STRUCT, was {}", schemaType);
+                    break;
+            }
+        }
+
+        return headers;
+    }
+
+    private Integer getPartitionNumber(SourceRecord sourceRecord) {
+        switch (partitionSetting) {
+            case PARTITION_NUMBER:
+                return getExplicitPartitionNumber(sourceRecord);
+            case PARTITION_KEY:
+                return getGeneratedPartitionNumber(sourceRecord);
+            default:
+                throw new IllegalArgumentException("Invalid partitionSetting provided " + partitionSetting);
+        }
+    }
+
+    private Integer getExplicitPartitionNumber(SourceRecord sourceRecord) {
+        try {
+            Integer partition = ((Struct) sourceRecord.value()).getInt32("partition_number");
+            assert partition != null : "partition_number is null";
+            return partition;
+        } catch (Exception|AssertionError e) {
+            throw new DataException("Unable to find partition_number in source record " + sourceRecord.toString(), e);
+        }
+    }
+    
+    private Integer getGeneratedPartitionNumber(SourceRecord sourceRecord) {
+        String partitionKey;
+        try {
+            partitionKey = ((Struct) sourceRecord.value()).getString(PARTITION_KEY_FIELD);
+            assert partitionKey != null : "partition_key is null";
+        } catch (Exception|AssertionError e) {
+            throw new DataException("Unable to find partition_key in source record " + sourceRecord.toString(), e);
+        }
+        try {
+            return Utils.toPositive(Utils.murmur2(partitionKey.getBytes())) % numberOfPartitionsInTargetTopic;
+        } catch (Exception e) {
+            throw new DataException("Unable to generate partition_number from partition_key " + partitionKey, e);
+        }
+    }
+
+    @java.lang.Override
+    public ConfigDef config() {
+        return CONFIG_DEF;
+    }
+
+    @java.lang.Override
+    public void close() {
+        // not in use
+    }
+
+    // Logging levels
+    public enum DebugLogLevel {
+        DEBUG, TRACE, INFO
+    }
+
+    private void DEBUG(String format, Object ...args) {
+        switch (this.debugLogLevel) {
+            case DEBUG:
+                LOGGER.debug(format, args);
+                break;
+            case TRACE:
+                LOGGER.trace(format, args);
+                break;
+            case INFO:
+                LOGGER.info(format, args);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid log level provided");
+        }
+    }
```

Comment:
All of this should be handled by the logger: you should continue using `LOGGER.debug` and we can configure what gets actually logged when running the task and/or Kafka Connect. I think you should just remove all of that logic, including the configuration parameter, basically. Happy to dig with you on how we configure Kafka Connect's task to output debug logs.

Reply:
my rationale: I don't want all of Kafka Connect's internal debug logs to drown out the SMT logs. That's why I'd prefer to have this additional flexibility

Reply:
You can do that by changing the logging level per logger class name: https://docs.confluent.io/platform/current/connect/logging.html#using-the-kconnect-api

Reply:
ooh nice 👍